### ISSUE TITLE
Slender object detection only has low PR and mAP

### ISSUE THREAD

**Issue by Yanhui-He** [ultralytics/yolov5#4030]
Yolov5 is a wonderful work in object detection. I used it in pedestrian detection and vehicle detection, which has great pweformence. However, when I try to use yolov5 to detect the pole-like object, such as light pole, traffic pole, wire pole, and sign pole, the model has a poor performence.

Note: I only want to detect single class that is pole.  

I have changed the ratio of width and height in ddateset.py (the max ar_thr is 63 in my own dataset):
`def box_candidates(box1, box2, wh_thr=1, ar_thr=100, area_thr=0.1, eps=1e-16): `

On the other hand, the custom data set has more than 9000 pole labels. So I think the dataset is enough to train the model.
The result is: Precison = 68.8, Recall = 36.0, mAP@.5 = 26.8. 

> (I use yolov5l6.yaml, the conf is 0.25 when I test the image.)

I also find a paper that report the slender detection: (https://arxiv.org/abs/2011.08529)

This problem has troubled me for a long time, please help me or give me some enlightenment.

**Comment by glenn-jocher** [comment]
@Yanhui-He you may need to relax the box candidate criteria here to allow higher aspect ratio labels. The default is 20, you could raise this to 100 for example for extremely thin objects:
https://github.com/ultralytics/yolov5/blob/b7e985e397e12ea2efd19bf3b6329028fb2a4c75/utils/augmentations.py#L266-L271

**Comment by glenn-jocher** [comment]
@Yanhui-He ah I see you already raise the limit. In this case you may simply need to train at higher resolution to capture more pixels in the short dimension, i.e. train at --img 1280 or 2560.

Remember also that YOLOv5 is designed for COCO-like objects, if your use case relies highly on high AR objects you may want to modify the convolutions in the C3 layers from 3x3 to alternating 1x9 and 9x1 for example.

**Comment by Yanhui-He** [comment]
@glenn-jocher I used --img-size 1472 to train the model, and test the model in 1920 resolution images. Moreover, I try to use DWConv to replace Conv module. However, the recall and mAP@.5 is less than 50%.

And I don't konw how to change the convolutions in the C3 layers. For example:
I changed the code in common.py class C3 module:

`self.cv1 = Conv(c1, c_, 1, 1)` >> `self.cv1 = Conv(c1, c_, (9, 1), 1)`

but that is wrong.

Oh, To improve the performence, I widen the pole's label width. (original label width * 2), which can improve the precision. Currently, the precision is 80% where the confidence is 0.5.

Meawhile, I changed the objectness in loss.py:

`tobj[b, a, gj, gi] = (1.0 - self.gr) + self.gr * iou.detach().clamp(0).type(tobj.dtype)` >> `tobj[b, a, gj, gi] = 1.0`

It can improve the object confidence when detecting and testing. Therefore, all of my result is testing in the 0.5 confidence.
